zulipterminal/ui_tools/boxes.py

```diff
@@ -230,15 +230,34 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         elif is_command_key('TAB', key):
             if len(self.contents) == 0:
                 return key
+            header = self.contents[0][0]
             # toggle focus position
-            if self.focus_position == 0 and self.to_write_box is None:
-                if self.contents[0][0].focus_col == 0:
-                    self.contents[0][0].focus_col = 1
-                    return key
+            if self.focus_position == 0:
+                if self.to_write_box is None:
+                    if header.focus_col == 0:
+                        stream_name = header[0].edit_text
+                        if not self.model.is_valid_stream(stream_name):
+                            self.view.set_footer_text("Invalid stream name", 3)
+                            return key
+
+                        header.focus_col = 1
+                        return key
+                    else:
+                        header.focus_col = 0
                 else:
-                    self.contents[0][0].focus_col = 0
+                    recipient_box = header.original_widget
+                    recipient_emails = recipient_box.edit_text.split(', ')
+                    invalid_emails = self.model.get_invalid_recipient_emails(
+                                                              recipient_emails)
+                    invalid_emails_str = ', '.join(invalid_emails)
+                    if invalid_emails:
+                        footer_text = "Invalid recipients - {}".format(
+                                                            invalid_emails_str)
```

Comment:
It seems reasonable to inline `invalid_emails_str` (it's only used in the conditional), and perhaps make `footer_text` more descriptive?

Reply:
How about "User(s) not found with email - abc@abc, abc@xyq" ?

Reply:
Or just "Could not find email(s) - abc@abc, abc@asd"?

Reply:
To clarify, I only meant the *name* of `footer_text` could be more descriptive.

I think the current text itself is fine, though you could add the parentheses to accommodate the single-recipient case.